[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion

mikaylagawarecki opened the pull request:
For above PR to parametrize existing `load_state_dict` tests

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118023
* #117167
* #118028

mikaylagawarecki committed:
[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion

[ghstack-poisoned]

mikaylagawarecki commented on test/test_modules.py:
```diff
@@ -866,65 +867,62 @@ def test_to(self, device, dtype, module_info, training, swap, set_grad):
                                                        requires_grad=False, training=training)
         torch.nn.utils.set_swap_module_params_on_conversion(swap)
 
-        try:
```
the diff is messed up here, all I did was remove the `try: finally` block

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

albanD commented on test/test_modules.py:
```diff
@@ -856,6 +856,7 @@ def test_errors(self, device, dtype, module_info, training):
     @modules([module for module in module_db if not module.is_lazy])
     @parametrize('swap', [True, False])
     @parametrize('set_grad', [True, False])
+    @wrapSwapTensorsTest(swap=None)
```
You don't need to pass in the swap argument here right?

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[easy] Add testing utilties for torch.nn.utils.set_swap_module_params_on_conversion"

For above PR to parametrize existing `load_state_dict` tests




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge
